SwiftReflector/SwiftInterfaceReflector/SyntaxDesugaringParser.cs

```diff
@@ -60,8 +75,8 @@ public override void ExitArray_type ([NotNull] Array_typeContext context)
 
 		public override void ExitDictionary_type ([NotNull] Dictionary_typeContext context)
 		{
-			var keyType = context.children [1].GetText ();
-			var valueType = context.children [3].GetText ();
+			var keyType = TypeText (context.children [1] as ParserRuleContext);
+			var valueType = TypeText (context.children [3] as ParserRuleContext);
```

Comment:
In this case it's safe - way the grammar is set up for `dictionary_type` is:
```
dictionary_type : OpLBracket type OpColon type OpRBracket
```
If this rule is matched, this is invariant: child [0] is always a left bracket, child [1] is always a type, child [2] is always a colon, child [3] is always a type and child [4] is always a right bracket.
children has to be the most general type, but since we know the invariance, we can make assumptions about the type and since `type` is a parser rule, `ParserRuleContext` will always be safe.
The only way this would break is if (1) the syntax for dictionary changed, which is unlikely, but possible but we would see it in unit test failures in other places and (2) I changed the `type` to be some other entity than a parser rule and the likelihood of that somewhere between slim and none, but leaning towards none mostly because it can be nothing but a parser rule. Even if I hoisted up a different entity - like suppose swift added different syntax for types depending on the context - it would still be a parser rule. The only other thing it could be is maybe a lexer rule and no. Just no.
Great question.